### PR TITLE
Add `simple_shader`

### PR DESCRIPTION
**Pull request by domesticmouse**
## Pre-launch Checklist

- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [X] I read the [Contributors Guide].
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-devrel channel on [Discord].

[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[Contributors Guide]: https://github.com/flutter/samples/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: simple_shader/shaders/simple.frag [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#version 460 core
+
+#include<flutter/runtime_effect.glsl>
+
+precision mediump float;
+
+layout(location=0)out vec4 fragColor;
+layout(location=0)uniform vec2 resolution;
```
Comment (jonahwilliams) [inline simple_shader/shaders/simple.frag:8]:
```suggestion
layout(location=0) uniform vec2 resolution;
```

File: simple_shader/shaders/simple.frag [hunk 2]
```diff
@@ -0,0 +1,25 @@
+#version 460 core
+
+#include<flutter/runtime_effect.glsl>
+
+precision mediump float;
+
+layout(location=0)out vec4 fragColor;
```
Comment (jonahwilliams) [inline simple_shader/shaders/simple.frag:7]: Place this after uniforms

```suggestion
layout(location=0) out vec4 fragColor;
```

File: simple_shader/shaders/simple.frag [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+layout(location=0)uniform vec2 resolution;
+
+vec3 flutterNavy = vec3(4,43,89)/255;
+vec3 flutterBlue = vec3(5,83,177)/255;
+vec3 flutterSky = vec3(2,125,253)/255;
+
+void main(){
+    vec2 st=FlutterFragCoord().xy/resolution.xy;
```
Comment (jonahwilliams) [inline simple_shader/shaders/simple.frag:15]:
```suggestion
    vec2 st = FlutterFragCoord().xy / resolution.xy;
```

File: simple_shader/shaders/simple.frag [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    
+    vec3 color = vec3(0.0);
+    vec3 percent = vec3((st.x+st.y)/2);
```
Comment (jonahwilliams) [inline simple_shader/shaders/simple.frag:18]:
```suggestion
    vec3 percent = vec3((st.x + st.y) / 2);
```

File: simple_shader/shaders/simple.frag [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+    color = mix(mix(flutterSky, flutterBlue, percent*2),
```
Comment (jonahwilliams) [inline simple_shader/shaders/simple.frag:20]:
```suggestion
    color = mix(mix(flutterSky, flutterBlue, percent * 2),
```

File: simple_shader/shaders/simple.frag [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+                mix(flutterBlue, flutterNavy, percent*2-1),
```
Comment (jonahwilliams) [inline simple_shader/shaders/simple.frag:21]:
```suggestion
                mix(flutterBlue, flutterNavy, percent * 2 - 1),
```

File: simple_shader/shaders/simple.frag [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+                step(0.5,percent));
```
Comment (jonahwilliams) [inline simple_shader/shaders/simple.frag:22]:
```suggestion
                step(0.5, percent));
```

File: simple_shader/shaders/simple.frag [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+    fragColor=vec4(color,1);
```
Comment (jonahwilliams) [inline simple_shader/shaders/simple.frag:24]:
```suggestion
    fragColor = vec4(color, 1);
```

**Review by jonahwilliams — APPROVED** [review]
Please format the shader source code. You can use tools like clang-format, or look for GLSL specific formatters.

**Review by craiglabenz — APPROVED** [review]
LGTM

**Pull request merged**